pelican_flickrtag/plugin.py

```diff
@@ -145,11 +175,15 @@ def replace_article_tags(generator):
             # Render the template
             replacement = template.render(context)
 
-            article._content = article._content.replace(match[0], replacement)
+            item._content = item._content.replace(match[0], replacement)
+
+
+def replace_article_tags(generator):
+    generic_replace(generator, 'article')
```

Comment:
I think you can use `generic_replace(generator, generator.articles)`. By doing that, you can remove the conditions and simplify the code. Probably, the downside would the the lack of a custom message of the photos found per article and page separately.